File: src/main/java/uk/claritygroup/model/UpdateMetrics.java
package uk.claritygroup.model;

import jakarta.validation.constraints.NotNull;
import lombok.Getter;

import java.util.Optional;
@Getter
public class UpdateMetrics {
    @NotNull
    private final String system;
    @NotNull
    private final String name;
    @NotNull
    private final Long date;
    private final Integer value;

    public UpdateMetrics(final String system, final String name, final Long date, final Integer value){
        this.system=system;
        this.name=name;
        this.date= date;
        this.value=Optional.ofNullable(value).isPresent()?Optional.of(value).get() :1;
    }
}

File: src/main/kotlin/uk/claritygroup/kotlin/model/request/UpdateMetrics.kt
package uk.claritygroup.kotlin.model.request

import jakarta.validation.constraints.NotNull

data class UpdateMetrics(
    @field:NotNull(message = "System must not be null")
    override val system: String?=null,
    @field:NotNull(message = "Name must not be null")
    override val name: String?=null,
    @field:NotNull(message = "Date must not be null")
    var date: Long? = null,
    override val value: Int? =1
): MetricsRequest(system,name,value)
